tools/dune_rule_gen/dune

```diff
@@ -0,0 +1,11 @@
+(library
+ (name coq_dune)
+ (modules :standard \ gen_rules)
+ (flags :standard -w +a-40-42)
```

Comment:
Can we not ignore these warnings and instead fix the code instead?

Reply:
There are reasonable warnings to ignore.

Reply:
Actually mabye @SkySkimmer , the thing is that we don't use the OCaml style I use here a lot in Coq, but for this style is critical. I don't remember the call well, but maybe it makes sense to reopen.

Style is a complex matter tho: I'm very happy with the dune style but of course it is my own pref.